verl/utils/dataset/sampler.py

```diff
@@ -21,16 +21,20 @@
 
 
 class AbstractSampler(Sampler[int]):
+    """Abstract interface for custom samplers."""
+
     @abstractmethod
     def __init__(
         self,
         data_source: Sized,
-        config: DictConfig,
+        data_config: DictConfig,
     ):
         pass
 
 
 class AbstractCurriculumSampler(AbstractSampler):
+    """Experimental interface for curriculum learning samplers."""
+
     @abstractmethod
-    def update(self, batch: DataProto) -> None:
+    def on_batch_end(self, batch: DataProto) -> None:
```

Comment:
Actually, I would prefer "update" as "on_batch_end" implicitly assumes that there will be a training loop, which doesn't really exists for async RL on the data sampling part.

Reply:
just changed it back